### ISSUE TITLE
Keep trying, Alice! Implement grant retries

### ISSUE THREAD

**Issue by KPrasch** [nucypher/nucypher#2480]
Even in the worst of conditions, Alice can continue to try granting as long as there are new nodes to sample.

**Comment by KPrasch** [comment]
@Bogdan#6932 - Do you think this issue needs to be open? Does the sampling logic introduced in  #2482 help here?
If not perhaps a simple retry loop at the right abstraction layer can resolve it.

**Comment by fjarri** [comment]
Depends on what is meant by "granting" here. There are retries at proposal stage, but no retries after the payment is made and some nodes claim it to be invalid. I guess this will change after #2530 ?

**Comment by cygnusv** [comment]
PRE not a priority now.
